google/cloud/spanner_v1/database.py

```diff
@@ -714,17 +775,24 @@ def execute_pdml():
                     query_options=query_options,
                     request_options=request_options,
                 )
-                method = functools.partial(
-                    api.execute_streaming_sql,
-                    metadata=metadata,
-                )
+
+                def wrapped_method(*args, **kwargs):
+                    partial_attempt.increment()
+                    method = functools.partial(
+                        api.execute_streaming_sql,
+                        metadata=self.metadata_with_request_id(
+                            partial_nth_request, partial_attempt.value, metadata
+                        ),
+                    )
+                    return method(*args, **kwargs)
 
                 iterator = _restart_on_unavailable(
-                    method=method,
+                    method=wrapped_method,
                     trace_name="CloudSpanner.ExecuteStreamingSql",
                     request=request,
                     transaction_selector=txn_selector,
                     observability_options=self.observability_options,
+                    attempt=begin_txn_attempt,
```

Comment:
I think that this should be `partial_attempt`. Retries of this RPC will not retry the `BeginTransaction` RPC, it will just continue to use the same transaction ID as the one that was returned by the original `BeginTransaction` RPC.

Reply:
Oooh, okay, that's very interesting and thanks for letting me know, let me make that update.

Reply:
I did however already pass in and increment partial_attempt inside wrapped_method. I'll instead remove that and increment them in the appropriate sites.